internal/target/service_list_ext_test.go

```diff
@@ -61,6 +63,106 @@ func TestService_List(t *testing.T) {
 	)
 	require.NoError(t, err)
 
+	t.Run("List validation", func(t *testing.T) {
+		t.Parallel()
+		t.Run("missing grants hash", func(t *testing.T) {
+			t.Parallel()
+			filterFunc := func(_ context.Context, t target.Target) (bool, error) {
+				return true, nil
+			}
+			tok, err := refreshtoken.New(ctx, time.Now(), time.Now(), resource.Session, []byte("some hash"), "some-id", time.Now())
+			require.NoError(t, err)
+			_, err = target.ListRefresh(ctx, nil, 1, filterFunc, tok, repo)
```

Comment:
You're calling `ListRefresh` in here instead of `List`

Reply:
Fixed, thanks